File: javascript/src/models/clickhouseConfigPatch.ts
// this file is @generated

export interface ClickhouseConfigPatch {
  url?: string;
  username?: string;
  password?: string;
  database?: string;
  tableName?: string;
}

export const ClickhouseConfigPatchSerializer = {
  _fromJsonObject(object: any): ClickhouseConfigPatch {
    return {
      url: object["url"],
      username: object["username"],
      password: object["password"],
      database: object["database"],
      tableName: object["tableName"],
    };
  },

  _toJsonObject(self: ClickhouseConfigPatch): any {
    return {
      url: self.url,
      username: self.username,
      password: self.password,
      database: self.database,
      tableName: self.tableName,
    };
  },
};

File: rust/src/models/clickhouse_config_patch.rs
// this file is @generated
#[allow(unused_imports)]
use js_option::JsOption;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct ClickhouseConfigPatch {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub url: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub username: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub password: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub database: Option<String>,

    #[serde(rename = "tableName")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub table_name: Option<String>,
}

impl ClickhouseConfigPatch {
    pub fn new() -> Self {
        Self {
            url: None,
            username: None,
            password: None,
            database: None,
            table_name: None,
        }
    }
}

impl Default for ClickhouseConfigPatch {
    fn default() -> Self {
        Self::new()
    }
}
